### ISSUE TITLE
Overhaul documentation

### ISSUE THREAD

**Issue by alexcrichton** [rust-lang/cargo#2580]
There's a few banner things that we need to clean up about the Cargo documentation:
- The look-and-feel is _very_ different than the main documentation (e.g. the book). Probably unnecessarily so?
- The organization is a little confusing today unfortunately. I've been told a few times that the content is all there, it's just not necessarily presented in a cohesive fashion that's easily queried.
- Perhaps we should just migrate all of the documentation to the book itself? Cargo is such a core tool in Rust that we probably don't want to have such standalone and somewhat-difficult-to-find documentation.

cc @steveklabnik, @brson, @aturon

**Comment by steveklabnik** [comment]
The second point is one I've tried several times, and each time, been like "idk". It's vanquished me. Heh.

The only problem I have with moving stuff to the book is that the book is already huge enough... it might make sense to reorganize Cargo's docs as its own book maybe? That would also help styling wise.

**Comment by alexcrichton** [comment]
Yeah I was basically just thinking that the docs here should be *very* easily accessible from the main docs (perhaps moreso than just a link) because you'll probably want to read all the cargo docs as well.

**Comment by nodakai** [comment]
A tutorial guide should be split off from the reference document.

The tutorial could be written to match the learning curve of a typical newcomer, with an emphasis on how to get started with Cargo. We might actually want to have two tutorials (or part I and II?), one for a crate user and another for a crate writer.

On the other hand the reference should list up each and every command line option and config item in a hierarchical way with an accurate description of its semantics. Rustdoc of the `cargo` command itself might or might not serve for this purpose (just a rough idea.)

**Comment by aturon** [comment]
cc @jonathandturner

**Comment by istankovic** [comment]
I would say this is one of the most important issues right now in cargo. I've always found it difficult to navigate cargo documentation. Several times I've tried to find the reference but somehow it seems the content is scattered all over the place. Maybe I'm just looking at wrong places, but it does feel frustrating and, honestly, I just resorted to using rustc directly instead of trying to find out how to tell cargo to do its thing. I think it would be good to compare this to the Rust documentation where it is trivial to get to The Book or to the reference, or really anything important, from the doc start page. Whoever did https://doc.rust-lang.org really did an excellent job.
